src/Data/Vec/Properties.agda

```diff
@@ -340,6 +342,15 @@ lookup-++ʳ []       ys       zero    = refl
 lookup-++ʳ []       (y ∷ xs) (suc i) = lookup-++ʳ [] xs i
 lookup-++ʳ (x ∷ xs) ys       i       = lookup-++ʳ xs ys i
 
+lookup-splitAt : ∀ m {n} (xs : Vec A m) (ys : Vec A n) i →
+                lookup (xs ++ ys) i ≡ [ lookup xs , lookup ys ]′
+                (Fin.splitAt m i)
+lookup-splitAt zero    []       ys i       = refl
+lookup-splitAt (suc m) (x ∷ xs) ys zero    = refl
+lookup-splitAt (suc m) (x ∷ xs) ys (suc i) = P.subst id
+  (P.cong (_ ≡_) (P.sym ([,]-map-comm (Fin.splitAt m i))))
```

Comment:
`P.cong (_ ≡_)` is simply an instance of `trans` and I'm pretty sure you can rewrite it without the `P.subst id`

Reply:
I don't think it's an instance of trans, or at least I don't see it.
What's happening here is that the type of the inductive call is `LHS == RHS'` but what it needs to be is `LHS == RHS`. Fortunately `RHS' == RHS` which we get by calling `[,]-map-comm`.
We then have to use the `P.cong (_ ≡_)` to get our desired `(LHS == RHS') == (LHS == RHS)`.
`P.subst id` merely turns this equality into a function so that we can get our desired result.
I would be very surprised if this can be made any more concise than what it is now.

Reply:
> EDIT: Nevermind, I see it now. Just tried it and it works! That's so much better than what it was before!

Haha, great to hear it :smile: it's always satisfying when it just clicks into place!